simulator/modules/core/perf_sim.cpp

```diff
@@ -99,6 +109,24 @@ auto get_rate( int total, float64 piece)
     return total != 0 ? ( piece / total * 100) : 0;
 }
 
+/* standard lines loggers implementation */
+template<typename ISA>
+void PerfSim<ISA>::start_dump_json() {
+    if (jsonout_enabled)
+        (jsonout()) << "[\n" <<
+        "\t{ \"type\": \"Stage\", \"id\": 0, \"description\": \"Fetch\" },\n" <<
+        "\t{ \"type\": \"Stage\", \"id\": 1, \"description\": \"Decode\" },\n" <<
+        "\t{ \"type\": \"Stage\", \"id\": 2, \"description\": \"Execute\" },\n" <<
+        "\t{ \"type\": \"Stage\", \"id\": 3, \"description\": \"Memory\" },\n" <<
+        "\t{ \"type\": \"Stage\", \"id\": 4, \"description\": \"Writeback\" }";
+}
+
+template<typename ISA>
+void PerfSim<ISA>::stop_dump_json() {
```

Comment:
It must be called in destructor, so file closing and proper finalization would be guaranteed.